src/agoradatatools/great_expectations/gx/plugins/expectations/expect_column_nested_field_values_to_be_mostly_not_null.py

```diff
@@ -0,0 +1,329 @@
+"Custom expectation rule that counts the percentage of nulls in a targeted field"
+
+from typing import Dict, Optional, List, Any
+import pandas as pd
+import json
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.execution_engine import (
+    ExecutionEngine,
+    PandasExecutionEngine,
+)
+from great_expectations.expectations.expectation import (
+    ColumnAggregateExpectation,
+    InvalidExpectationConfigurationError,
+)
+from great_expectations.expectations.metrics import (
+    ColumnAggregateMetricProvider,
+    column_aggregate_value,
+)
+
+METRIC_NAME = "column.nested_object_not_null_ratio"
+
+# This method implements the core logic for the PandasExecutionEngine
+class ColumnNestedObjectNotNull(ColumnAggregateMetricProvider):
+    """A custom Great Expectations metric that calculates the proportion of non-null values
+    for a specified key across nested lists of dictionaries in a column."""
+
+    metric_name = METRIC_NAME
+    value_keys = ("non_null_threshold", "target_field")
+
+    @staticmethod
+    def safe_parse(value: str) -> List[Dict[str, Any]]:
+        """
+        Load a JSON string and return a list of dictionaries.
+        If the input is not a valid JSON, return an empty list.
+
+        Parameters:
+            value[str]: the json string to be parsed. If input is not a valid json string, return an empty list
+
+        Returns:
+            Parsed json object or fall back to an empty list
+        """
+        try:
+            parsed = json.loads(value)
+            if isinstance(parsed, list):
+                return parsed
+            else:
+                return []
+        # Fallback if it's not valid JSON like "null"
+        except (json.JSONDecodeError, TypeError):
```

Comment:
Should anything be logged here on this failure? As this is it will silently fail

Reply:
A check to verify that the data is actually null, in those cases we can probably ignore the exception. But for any other cases we either want to log it as an exception, or let the except be raised and fail processing.

Reply:
Good idea. I updated the code so that it falls back to [] when the value is null, but raise other errors.